### PR TITLE
Added ignoreInvalidDefaults parameter to codegen main method

### PR DESCRIPTION
**Pull request by Falmarri**
Was previously hard coded to false to false

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @Falmarri,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://www.typesafe.com/contribute/cla](http://www.typesafe.com/contribute/cla)

File: slick-codegen/src/main/scala/slick/codegen/SourceCodeGenerator.scala [hunk 1]
```diff
@@ -90,9 +91,11 @@ object SourceCodeGenerator {
       case uri :: outputDir :: Nil =>
         run(new URI(uri), Some(outputDir))
       case profile :: jdbcDriver :: url :: outputDir :: pkg :: Nil =>
-        run(profile, jdbcDriver, url, outputDir, pkg, None, None)
+        run(profile, jdbcDriver, url, outputDir, pkg, None, None, false)
       case profile :: jdbcDriver :: url :: outputDir :: pkg :: user :: password :: Nil =>
-        run(profile, jdbcDriver, url, outputDir, pkg, Some(user), Some(password))
+        run(profile, jdbcDriver, url, outputDir, pkg, Some(user), Some(password), false)
+      case  profile:: jdbcDriver :: url :: outputDir :: pkg :: user :: password :: ignoreInvalidDefaults :: Nil =>
+        run(profile, jdbcDriver, url, outputDir, pkg, Some(user), Some(password), Try(ignoreInvalidDefaults.toBoolean).getOrElse(false))
```
Comment (cvogt) [inline slick-codegen/src/main/scala/slick/codegen/SourceCodeGenerator.scala]: not a fan of silently ignoring errors as in `Try(...).getOrElse(...)`. Let's let users know something went wrong. The easiest way would be simply throwing the exception, which I would be ok with.

**Comment by lsierant** [comment]
Hey @Falmarri, are you going to proceed with this? It would be nice to have it in an official release.

**Comment by Falmarri** [comment]
I have? I already filled out the CLA. What else do you need from me?

**Comment by kentongray** [comment]
i really wish someone on the slick team would take this great request from @Falmarri (it would solve a problem I'm having right now!)

**Comment by kentongray** [comment]
ping @lsierant anything @Falmarri  can do to get this in?

**Comment by cvogt** [comment]
@Falmarri @kentongray having time to look into this now. You use default value false. In other places we use the default value true for this, because for something like the code generator it seems to be the better user experience to me to do something but potentially leave out default values rather than crashing. I didn't realize it had been hard coded to a false call at some point. WDYT about changing it to true and adjusting the default values as well? In any case, this is an improvement as is, so I am merging.

**Comment by cvogt** [comment]
this seems to validate the assumption, that `true` may be a better default value: https://github.com/slick/slick/issues/1146

**Comment by Falmarri** [comment]
@cvogt That's fine with me. I didn't want to make assumptions and change the default behavior on you guys.

**Comment by stewSquared** [comment]
@cvogt When will this be published?

**Comment by cvogt** [comment]
@stewSquared as soon as one of the core commiters finds the time to release a milestone. We should totally do that asap. I'll get the ball rolling.

**Comment by whenceforth** [comment]
Any chance you could prioritize that milestone?  Not having this is a blocker for adopting Slick for legacy databases.  Well, at least 2 of them, @Falmarri's  and mine :-)

**Pull request merged**